packages/mcp-docs-server/src/utils.ts

```diff
@@ -12,3 +13,143 @@ export function fromPackageRoot(relative: string) {
 
 // can't use console.log() because it writes to stdout which will interfere with the MCP Stdio protocol
 export const log = console.error;
+
+async function* walkMdxFiles(dir: string): AsyncGenerator<string> {
+  const entries = await fs.readdir(dir, { withFileTypes: true });
```

Comment:
I wonder if we should do an in memory cache with a map or something to save memory here

Reply:
yeah i'll look into adding something for this